package/base-files/files/lib/upgrade/common.sh

```diff
@@ -267,7 +295,7 @@ get_partitions() { # <device> <filename>
 				local type="$1"
 				local lba="$(( $(hex_le32_to_cpu $4) * 0x100000000 + $(hex_le32_to_cpu $3) ))"
 				local end="$(( $(hex_le32_to_cpu $6) * 0x100000000 + $(hex_le32_to_cpu $5) ))"
-				local num="$(( $end - $lba ))"
+				local num="$(( $end - $lba + 1 ))"
```

Comment:
Why the index-off-by-one change?

Reply:
To match the boundaries shown by a `gdisk -l`

Reply:
If that's the case, it might make more sense to fix that as a separate issue/PR.

Reply:
I still have included it here.

I guess I also need to cleanup the commits although I am not sure how to best split it up. I'd welcome any suggestions.

@adschm @pprindeville ping